### PR TITLE
(maint) Move puppet-agent tests out of puppet

### PR DESCRIPTION
**Pull request by melissa**
This test is specific to setup that happens as a part of the
puppet-agent package. We should have this test live as a part of the
puppet-agent acceptance tests

### CODE REVIEW

**Comment by melissa** [comment]
jenkins please test this on windows2012r2-64

**Comment by melissa** [comment]
corresponding puppet change: https://github.com/puppetlabs/puppet/pull/7124

**Comment by melissa** [comment]
It looks like this ran, but I'm not seeing any output

from the puppet-agent pipeline:
```
Begin tests/env_windows_installdir_fact.rb

PA-466: Ensure env_windows_installdir fact is present and correct
Begin teardown
End teardown
tests/env_windows_installdir_fact.rb passed in 0.00 seconds
```

from the puppet 5.5.x pipeline:
```
Begin tests/windows/env_windows_installdir_fact.rb

PA-466: Ensure env_windows_installdir fact is present and correct
  
  * test for presence/accuracy of fact on kxrmw3tbelo5q9u.delivery.puppetlabs.net
    
    kxrmw3tbelo5q9u.delivery.puppetlabs.net (kxrmw3tbelo5q9u.delivery.puppetlabs.net) 03:56:09$ cmd.exe /c puppet facts --render-as json
      {"name":"kxrmw3tbelo5q9u.delivery.puppetlabs.net","values":{"aio_agent_version":"5.5.6.125","architecture":"x64","dhcp_servers":{"Ethernet0":"10.32.22.9","system":"10.32.22.9"},"dmi":{"manufacturer":"VMware, Inc.","product":{"name":"VMware7,1","serial_number":"VMware-42 1a dc 43 76 f4 d4 1e-81 9d d7 81 71 62 02 be"}},"domain":"delivery.puppetlabs.net","env_windows_installdir":"C:\\Program Files\\Puppet Labs\\Puppet","facterversion":"3.11.5","fqdn":"kxrmw3tbelo5q9u.delivery.puppetlabs.net","hardwareisa":"x64","hardwaremodel":"x86_64","hostname":"kxrmw3tbelo5q9u","hypervisors":{"vmware":{}},"id":"KXRMW3TBELO5Q9U\\Administrator","identity":{"privileged":true,"user":"KXRMW3TBELO5Q9U\\Administrator"},"interfaces":"Ethernet0","ipaddress":"10.16.120.36","ipaddress6":"fe80::5dc7:48a4:8f9f:b97f%12","ipaddress6_Ethernet0":"fe80::5dc7:48a4:8f9f:b97f%12","ipaddress_Ethernet0":"10.16.120.36","is_virtual":true,"kernel":"windows","kernelmajversion":"6.3","kernelrelease":"6.3.9600","kernelversion":"6.3.9600","macaddress":"00:50:56:9A:8F:C6","macaddress_Ethernet0":"00:50:56:9A:8F:C6","manufacturer":"VMware, Inc.","memory":{"system":{"available":"3.27 GiB","available_bytes":3514208256,"capacity":"18.16%","total":"4.00 GiB","total_bytes":4293947392,"used":"743.62 MiB","used_bytes":779739136}},"memoryfree":"3.27 GiB","memoryfree_mb":3351.41015625,"memorysize":"4.00 GiB","memorysize_mb":4095.02734375,"mtu_Ethernet0":1500,"netmask":"255.255.240.0","netmask6":"ffff:ffff:ffff:ffff::","netmask6_Ethernet0":"ffff:ffff:ffff:ffff::","netmask_Ethernet0":"255.255.240.0","network":"10.16.112.0","network6":"fe80::%12","network6_Ethernet0":"fe80::%12","network_Ethernet0":"10.16.112.0","networking":{"dhcp":"10.32.22.9","domain":"delivery.puppetlabs.net","fqdn":"kxrmw3tbelo5q9u.delivery.puppetlabs.net","hostname":"kxrmw3tbelo5q9u","interfaces":{"Ethernet0":{"bindings":[{"address":"10.16.120.36","netmask":"255.255.240.0","network":"10.16.112.0"}],"bindings6":[{"address":"fe80::5dc7:48a4:8f9f:b97f%12","netmask":"ffff:ffff:ffff:ffff::","network":"fe80::%12"}],"dhcp":"10.32.22.9","ip":"10.16.120.36","ip6":"fe80::5dc7:48a4:8f9f:b97f%12","mac":"00:50:56:9A:8F:C6","mtu":1500,"netmask":"255.255.240.0","netmask6":"ffff:ffff:ffff:ffff::","network":"10.16.112.0","network6":"fe80::%12"}},"ip":"10.16.120.36","ip6":"fe80::5dc7:48a4:8f9f:b97f%12","mac":"00:50:56:9A:8F:C6","mtu":1500,"netmask":"255.255.240.0","netmask6":"ffff:ffff:ffff:ffff::","network":"10.16.112.0","network6":"fe80::%12","primary":"Ethernet0"},"operatingsystem":"windows","operatingsystemmajrelease":"2012 R2","operatingsystemrelease":"2012 R2","os":{"architecture":"x64","family":"windows","hardware":"x86_64","name":"windows","release":{"full":"2012 R2","major":"2012 R2"},"windows":{"system32":"C:\\Windows\\system32"}},"osfamily":"windows","path":"C:\\Program Files\\Puppet Labs\\Puppet\\facter/bin;C:\\Program Files\\Puppet Labs\\Puppet\\facter/../puppet/bin;C:\\Program Files\\Puppet Labs\\Puppet\\puppet\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\facter\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\hiera\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\mcollective\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\sys\\ruby\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\sys\\tools\\bin;PATH;C:\\Program Files (x86)\\Puppet Labs\\Puppet\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\bin;C:\\cygwin64\\bin;C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0;C:\\Program Files\\Git\\cmd;C:\\Packer\\SysInternals","physicalprocessorcount":2,"processor0":"Intel(R) Xeon(R) CPU E5-2697 v4 @ 2.30GHz","processor1":"Intel(R) Xeon(R) CPU E5-2697 v4 @ 2.30GHz","processorcount":2,"processors":{"count":2,"isa":"x64","models":["Intel(R) Xeon(R) CPU E5-2697 v4 @ 2.30GHz","Intel(R) Xeon(R) CPU E5-2697 v4 @ 2.30GHz"],"physicalcount":2},"productname":"VMware7,1","puppetversion":"5.5.7","ruby":{"platform":"x64-mingw32","sitedir":"C:/Program Files/Puppet Labs/Puppet/sys/ruby/lib/ruby/site_ruby/2.4.0","version":"2.4.4"},"rubyplatform":"x64-mingw32","rubysitedir":"C:/Program Files/Puppet Labs/Puppet/sys/ruby/lib/ruby/site_ruby/2.4.0","rubyversion":"2.4.4","serialnumber":"VMware-42 1a dc 43 76 f4 d4 1e-81 9d d7 81 71 62 02 be","system32":"C:\\Windows\\system32","system_uptime":{"days":0,"hours":2,"seconds":7826,"uptime":"2:10 hours"},"timezone":"Coordinated Universal Time","uptime":"2:10 hours","uptime_days":0,"uptime_hours":2,"uptime_seconds":7826,"virtual":"vmware","clientcert":"kxrmw3tbelo5q9u.delivery.puppetlabs.net","clientversion":"5.5.7","clientnoop":false},"timestamp":"2018-09-27T03:56:09.193667000+00:00","expiration":"2018-09-27T04:26:09.193667000+00:00"}
    
    kxrmw3tbelo5q9u.delivery.puppetlabs.net (kxrmw3tbelo5q9u.delivery.puppetlabs.net) executed in 1.40 seconds
Begin teardown
End teardown
tests/windows/env_windows_installdir_fact.rb passed in 1.41 seconds
```

**Comment by joshcooper** [comment]
Could you also move `ensure_puppet-agent_paths.rb` and `ensure_version_file.rb` to puppet-agent?

**Comment by melissa** [comment]
jenkins please test this on windows2012r2-64

**Comment by melissa** [comment]
@kevpl @mchllweeks though this is failing, on windows, it looks like we're not actually running any of the tests. The output is all along the lines of
```
Begin tests/ensure_puppet-agent_paths.rb

PUP-4033: Ensure aio path spec is honored

* test configprint outputs

* test puppet genconfig entries

* test puppet config paths exist

* test puppet binaries exist
Begin teardown
```

It looks like we do have testing output for redhat and ubuntu, just not windows. Any thoughts as to why?

File: acceptance/tests/ensure_puppet-agent_paths.rb [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    {:name => :route_file,      :expected => "#{confdir}/routes.yaml"},
+    {:name => :ssldir,          :expected => "#{confdir}/ssl",            :installed => :dir},
+    {:name => :hiera_config,    :expected => "#{confdir}/hiera.yaml"},
+
+    # vardir
+    {:name => :vardir,          :expected => "#{vardir}",                 :installed => :dir},
+    {:name => :bucketdir,       :expected => "#{vardir}/bucket"},
+    {:name => :clientyamldir,   :expected => "#{vardir}/client_yaml",     :installed => :dir},
+    {:name => :client_datadir,  :expected => "#{vardir}/client_data",     :installed => :dir},
+    {:name => :clientbucketdir, :expected => "#{vardir}/clientbucket",    :installed => :dir},
```
Comment (melissa) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: It looks like the same is true for all three of these, `client_yaml`, `client_data`, and `clientbucket`

Comment (melissa) [reply acceptance/tests/ensure_puppet-agent_paths.rb]: Are these three lines safe to remove, or should I figure out why these directories aren't being created?

**Comment by melissa** [comment]
jenkins please test this on windows2012r2-64a

**Comment by melissa** [comment]
jenkins please test this on windows2012r2-64a

File: acceptance/tests/ensure_puppet-agent_paths.rb [hunk 2]
```diff
@@ -66,9 +66,6 @@ def config_options(agent)
     # vardir
     {:name => :vardir,          :expected => "#{vardir}",                 :installed => :dir},
     {:name => :bucketdir,       :expected => "#{vardir}/bucket"},
-    {:name => :clientyamldir,   :expected => "#{vardir}/client_yaml",     :installed => :dir},
-    {:name => :client_datadir,  :expected => "#{vardir}/client_data",     :installed => :dir},
-    {:name => :clientbucketdir, :expected => "#{vardir}/clientbucket",    :installed => :dir},
```
Comment (melissa) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: @joshcooper @jhelwig I just wanted to draw your attention to the fact that I removed these lines. They were causing the acceptance runs to fail, and it looks like we don't create any of these directories as a part of the package installation process. They are set as defaults in puppet, and previously were passing as a part of the puppet acceptance suite.

Comment (joshcooper) [reply acceptance/tests/ensure_puppet-agent_paths.rb]: @melissa that makes sense. Those directories are created by puppet via its settings catalog magic, and probably only get created when running `puppet agent`. I'm 👍 to only test things that we expect the MSI package to create.

Comment (melissa) [reply acceptance/tests/ensure_puppet-agent_paths.rb]: this will be for all distros, not just msis, but that sounds good to me!

**Pull request merged**